pkg/acceptor/quic_acceptor.go

```diff
@@ -0,0 +1,244 @@
+func (a *QuicAcceptor) Accept() (quic.Connection, error) {
+	if a.listener == nil { // Correct comparison
+		return nil, ErrListenerNotInitialized
+	}
+
+	conn, err := a.listener.Accept(context.Background())
+	if err != nil {
+		return nil, err
+	}
+
+	return conn, nil
+}
+
+// Close closes the QUIC listener
+func (a *QuicAcceptor) Close() error {
+	if a.listener != nil { // Correct comparison
+		return a.listener.Close()
+	}
+	return nil
+}
+
+// EnableProxyProtocol not implemented for QUIC, keep as No-op or implement as needed
+func (a *QuicAcceptor) EnableProxyProtocol() {
+	// No-op: Implement this method if needed for Proxy Protocol support
+}
+
+func (a *QuicAcceptor) IsRunning() bool {
+	return a.running
+}
+
+func (a *QuicAcceptor) GetConfiguredAddress() string {
+	return a.addr
+}
+
+func (a *QuicAcceptor) ListenAndServe() {
+	// Start the QUIC listener
+	if err := a.Listen(); err != nil {
+		fmt.Printf("Failed to start QUIC listener: %s\n", err)
+		return
+	}
+
+	// Loop to continuously accept connections
+	for a.IsRunning() {
+		conn, err := a.Accept()
+		if err != nil {
+			if errors.Is(err, ErrListenerNotInitialized) {
+				fmt.Println("Listener not initialized")
+				continue
+			}
+			fmt.Printf("Failed to accept connection: %s\n", err)
+			continue
+		}
+
+		// Send the connection to the channel for further processing
+		go func(c quic.Connection) {
+			playerConn := NewQuicConnWrapper(c)
+			a.connChan <- playerConn
+		}(conn)
+	}
+}
+
+// QuicConnWrapper is a wrapper for a QUIC connection, allowing the use of deadlines
+type QuicConnWrapper struct {
+	conn quic.Connection
+}
+
+// NewQuicConnWrapper creates a new wrapper for a QUIC connection
+func NewQuicConnWrapper(conn quic.Connection) *QuicConnWrapper {
+	return &QuicConnWrapper{conn: conn}
+}
+
+// Read reads data from the QUIC connection
+func (q *QuicConnWrapper) Read(p []byte) (int, error) {
+	stream, err := q.conn.AcceptStream(context.Background())
+	if err != nil {
+		return 0, err
+	}
+
+	return stream.Read(p)
+}
+
+// Write writes data to the connection with a defined deadline
+func (q *QuicConnWrapper) Write(p []byte) (int, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second) // 5 seconds timeout as an example
+	defer cancel()
+
+	stream, err := q.conn.OpenStreamSync(ctx)
+	if err != nil {
+		return 0, err
+	}
+
+	return stream.Write(p)
+}
+
+// Close closes the QUIC connection
+func (q *QuicConnWrapper) Close() error {
+	return q.conn.CloseWithError(0, "closed")
+}
+
+// LocalAddr returns the local address of the connection
+func (q *QuicConnWrapper) LocalAddr() net.Addr {
+	return q.conn.LocalAddr()
+}
+
+// RemoteAddr returns the remote address of the connection
+func (q *QuicConnWrapper) RemoteAddr() net.Addr {
+	return q.conn.RemoteAddr()
+}
+
+// GetNextMessage reads the next message available in the QUIC stream
+func (q *QuicConnWrapper) GetNextMessage() (b []byte, err error) {
+	// Read data from the stream
```

Comment:
can't you leverage io utilities that go provides in this method, just like we do for the tcp acceptor?
https://github.com/topfreegames/pitaya/blob/main/pkg/acceptor/tcp_acceptor.go#L55-L74

Reply:
also, you seem to be assuming the maximum size of a message will be 1024 bytes, which is inaccurate

Reply:
I had initially left the limit fixed as a testing approach, but I tested it with io utilities, and it worked fine. Thanks for pointing that out; it was something I had overlooked